diff.py

```diff
@@ -2251,10 +2251,16 @@ def diff_sameline(old_line: Line, new_line: Line) -> None:
             newfields = newfields[:-1]
             oldfields = oldfields[:-1]
         else:
-            # If the last field has a parenthesis suffix, i.e. "0x38(r7)" or "%lo(.data)" or "%hi(.rodata + 0x10)"
+            # If the last field has a parenthesis suffix, i.e. "0x38(r7)"
             # we split that part out to make it a separate field
-            newfields = newfields[:-1] + newfields[-1].split("(")
-            oldfields = oldfields[:-1] + oldfields[-1].split("(")
+            # however, we don't split if it has a proceeding %hi/%lo  i.e."%lo(.data)" or "%hi(.rodata + 0x10)"
+            if len(oldfields) > 1:
```

Comment:
no need for this check, from what I can see

Reply:
oh guess you are right